### PR TITLE
Ocelots don't take fall damage

### LINKED ISSUE #4157: ocelot's getting damage when falling
Client version:   1.12.2
Server OS:        Windows 7
Commit id:        https://github.com/cuberite/cuberite/commit/ec9e0eecf616bb190e4d0be76f1c120cac9eb60b

### Expected behavior
ocelot doesn't get damage when it falls

### Actual behavior
ocelot dies or gets some damage

### Steps to reproduce the behavior
1. place some blocks
2. spawn ocelot
3. make him fall

sorry for my bad English)

### PR DESCRIPTION
**Pull request by Bond-009**
Fixes #4157

### CODE REVIEW

File: src/Mobs/Ocelot.cpp [hunk 1]
```diff
@@ -218,3 +218,8 @@ bool cOcelot::IsCatSittingOnBlock(cWorld * a_World, Vector3d a_BlockPosition)
 
 
 
+
+void cOcelot::HandleFalling(void)
+{
+	// empty - Ocelots don't take fall damage
```
Comment (peterbell10) [inline src/Mobs/Ocelot.cpp]: This function did a lot more than just do damage.  A better way to do damage immunity this is to override `DoTakeDamage` and not call `super::DoTakeDamage` for `dtFalling`.

Comment (bearbin) [reply src/Mobs/Ocelot.cpp]: A similar method was used in #4136, so you may want to look there for any guidance.

Comment (Bond-009) [reply src/Mobs/Ocelot.cpp]: Should I change the fall damage logic for chickens too? (It's done the same way as this PR)

Comment (peterbell10) [reply src/Mobs/Ocelot.cpp]: I think that would be best.

Comment (Bond-009) [reply src/Mobs/Ocelot.cpp]: Done

**Pull request merged**